src/endpoint/s3/s3_rest.js

```diff
@@ -118,9 +119,7 @@ async function handle_request(req, res) {
     usage_report.s3_usage_info.total_calls += 1;
     usage_report.s3_usage_info[op_name] = (usage_report.s3_usage_info[op_name] || 0) + 1;
 
-
-
-    if (req.query && req.query.versionId) {
+    if (s3_utils.parse_version_id(req.query?.versionId)) {
```

Comment:
I don't think this should be changed. the entire point in this code block was to prevent versioning for caching bucket. I agree that this check as a whole should have been added differently, perhaps just in a separate function than inline like that in the rest api flow, but still, we do not need to affect it with validation of the version id.